### PR TITLE
Refactor/questionnaire

### PR DESCRIPTION
**Pull request by lustigy96**

### CODE REVIEW

File: src/pages/QuestionnaireSchemaPage/components/SchemasTable/SchemasTable.jsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  ];
+
+  const getIntendedRole = (rolesArr) => {
+    if (!rolesArr.length) {
+      return '-';
+    }
+    if (rolesArr.includes(Role.Malshab) && rolesArr.length === 1) {
+      return t('roles.malshab');
+    }
+    if (!rolesArr.includes('MALSHAB')) {
```
Comment (carmeltsadok) [inline src/pages/QuestionnaireSchemaPage/components/SchemasTable/SchemasTable.jsx:51]: use `Role.Malshab`

File: src/pages/QuestionnaireSchemaPage/components/config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    [QuestionType.linearScale]: ['min', 'max'],
+  },
+  emptyQuestionDefFields: {
+    title: '',
+    options: [],
+    isShort: false,
+    hasOther: false,
+    description: '',
+    min: { tag: '', value: 0 },
+    max: { tag: '', value: 5 },
```
Comment (carmeltsadok) [inline src/pages/QuestionnaireSchemaPage/components/config.js:19]: required ?!?!?!?!?!?!?!>1

File: src/services/node.service.js [hunk 3]
```diff
@@ -7,14 +8,16 @@ const headers = {
...
     return data;
-  }
-  static async getNodeById(id) {
+  }, { maxAge: 10000 });
+
+  static getNodeById = mem(async (id) => {
     const { data } = await axios.get(`${config.apiUri}/api/node/${id}`, { headers });
     return data;
-  }
+  }, { maxAge: 10000 });
```
Comment (carmeltsadok) [inline src/services/node.service.js:20]: add to appConf

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/pages/QuestionnaireSchemaPage/components/SchemasTable/SchemasTable.jsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        }
+      }
+      setAllQuestionnaireRows([...tmpAllQuestionnaireRows]);
+      setQuestionnaireToShow([...tmpQuestionnaireRowsShow]);
+    }
+  }, [questionnaireToEdit]);
+
+  const table = questionnaireRowsToShow.length
+    ? (
+      <div className={`${classes.root}`}>
```
Comment (carmeltsadok) [inline src/pages/QuestionnaireSchemaPage/components/SchemasTable/SchemasTable.jsx:177]:
```suggestion
      <div className={classes.root}>
```

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — APPROVED** [review]

**Review by maykremer — APPROVED** [review]

**Pull request merged**